numpy/ma/core.pyi

```diff
@@ -341,7 +341,7 @@ def getmask(a): ...
 get_mask = getmask
 
 def getmaskarray(arr): ...
-def is_mask(m): ...
+def is_mask(m: object) -> TypeIs[NDArray[bool_]]: ...
```

Comment:
Hmm, technically this isn't always true:

https://github.com/numpy/numpy/blob/bbdfd653f9c158992f5b18586857461629cdb205/numpy/ma/core.py#L1584-L1587

So it's sufficient for `m` to be a `dtype` with `type: type[np.bool_]`, which isn't necessarily a `ndarray`. 

I'm not sure how common this is in practice though 🤔 So maybe we can get away with leaving it like this, perhaps accompanied by a comment to open an issue if it's causing problems.